src/www/app/outline_server.ts

```diff
@@ -50,21 +52,50 @@ export class OutlineServer implements PersistentServer {
   }
 
   get name() {
-    return this.config.name || this.config.host || '';
+    return this.config.name || this.tunnel.config ?.name || this.host || '';
   }
 
   set name(newName: string) {
     this.config.name = newName;
+    if (this.tunnel.config) {
+      this.tunnel.config.name = newName;
+    }
   }
 
   get host() {
-    return this.config.host;
+    if (this.tunnel.config) {
+      return `${this.tunnel.config.host}:${this.tunnel.config.port}`;
+    }
+    // TODO(alalama): refine which components of the source URL to show.
+    return this.config.source ?.url || '';
   }
 
   async connect() {
     try {
+      if (this.config.source) {
```

Comment:
Let's move this logic to get the config to a ConfigSource object, with a getConfigForNewConnection().
This will separate the concerns. The OutlineServer does not have to know how the config is obtained.